### PR TITLE
Connect Wallet button is now a dropdown menu

### PR DESCRIPTION
**Pull request by krisbitney**
This PR also refactors the Header

Closes https://github.com/GoodDollar/GoodCollective/issues/30

### CODE REVIEW

File: packages/app/src/components/Header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+function Header(): JSX.Element {
+  const { address } = useAccount();
+  const [isDesktopResolution] = useMediaQuery({
+    minWidth: 612,
+  });
+  const { navigate } = useCrossNavigate();
+
+  return (
+    <View style={styles.headerOverlay}>
+      {isDesktopResolution && (
```
Comment (sirpy) [inline packages/app/src/components/Header/Header.tsx:20]: @L03TJ3 show @krisbitney how to use native-base to make this DRY

File: packages/app/src/components/Header/ConnectWalletMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  function connectorLogo(name: string): string {
+    switch (name) {
+      case 'MetaMask':
+        return metamaskLogoUri;
+      case 'WalletConnect':
+        return walletConnectLogoUri;
+      default:
+        return WebIconUri;
+    }
+  }
```
Comment (sirpy) [inline packages/app/src/components/Header/ConnectWalletMenu.tsx:28]: why aren't we using the regular web3modal widget?

File: packages/app/src/components/Header/RotatingArrowIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useRef } from 'react';
+import { Animated } from 'react-native';
+import { backIconUri } from './assets';
```
Comment (sirpy) [inline packages/app/src/components/Header/RotatingArrowIcon.tsx:3]: why are svgs not in simple svg files?

Comment (krisbitney) [reply packages/app/src/components/Header/RotatingArrowIcon.tsx:3]: react native doesn't natively support svg files. I can look into libraries to handle this later if there is time, but right now we aren't expanding the scope to include code style preferences

File: packages/app/src/components/Header/ConnectedAccountDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+  const [tokenBalance, setTokenBalance] = useState<string>('0');
+  const [ensName, setEnsName] = useState<any>('');
+
+  // TODO: what token is this? Will the token always be the same?
+  useEffect(() => {
+    if (!address) return;
+    fetchBalance({
+      address: address,
+      chainId: chain?.id,
+      token: '0x471EcE3750Da237f93B8E339c536989b8978a438',
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/Header/ConnectedAccountDisplay.tsx:31]: I don't know. we'll work it out in due time.

File: packages/app/src/components/Header/ConnectWalletMenu.tsx (hunk 2 above)
Comment (krisbitney) [reply packages/app/src/components/Header/ConnectWalletMenu.tsx:28]: I was told native app support is not within scope, and this was the quicker way to get it done

File: packages/app/src/components/Header/ConnectedAccountDisplay.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/app/src/components/Header/ConnectedAccountDisplay.tsx:31]: please note that this PR is not attempting to fix every issue that might be present in the header

**Pull request merged**